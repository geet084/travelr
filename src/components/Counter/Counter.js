import React from 'react';
import CountUp from 'react-countup';

export const Counter = ({ start, end, decimal = 0 }) => {
  //IGNORING COUNTUP PACKAGE FOR TESTING ... COUNTUPREF IS NOT TESTABLE
  /* istanbul ignore next line */
  return (
    <CountUp
      start={start}
      end={end}
      decimals={decimal}
      duration='1000000'
      delay={0}
      useEasing={false}
      separator=','
    >
      {({ countUpRef }) => <span ref={countUpRef} />}
    </CountUp>
  )
}

export default Counter;