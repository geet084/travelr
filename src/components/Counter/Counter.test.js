import React from 'react';
import { shallow } from 'enzyme';
import { Counter } from './Counter';

describe('Counter', () => {
  let wrapper;
  const mockProps = {
    start: 4,
    end: 19
  };
  wrapper = shallow(<Counter {...mockProps} />);

  it('should match initial snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});