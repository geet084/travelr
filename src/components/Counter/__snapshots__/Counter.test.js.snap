// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match initial snapshot 1`] = `
<CountUp$$1
  decimal="."
  decimals={0}
  delay={0}
  duration="1000000"
  easingFn={null}
  end={19}
  formattingFn={null}
  onEnd={[Function]}
  onPauseResume={[Function]}
  onReset={[Function]}
  onStart={[Function]}
  onUpdate={[Function]}
  prefix=""
  redraw={false}
  separator=","
  start={4}
  suffix=""
  useEasing={false}
>
  <Component />
</CountUp$$1>
`;
